lib/TopSitesFeed.jsm

```diff
@@ -76,17 +68,20 @@ this.TopSitesFeed = class TopSitesFeed {
     this._storage = this.store.dbStorage.getDbTable("sectionPrefs");
     this.refresh({broadcast: true});
     Services.obs.addObserver(this, "browser-search-engine-modified");
+    this._currentSearchHostname = getShortURLForCurrentSearch();
   }
 
   uninit() {
     PageThumbs.removeExpirationFilter(this);
     Services.obs.removeObserver(this, "browser-search-engine-modified");
+    this._currentSearchHostname = null;
   }
 
   observe(subj, topic, data) {
     // We should update the current top sites if the search engine has been changed since
     // the search engine that gets filtered out of top sites has changed.
-    if (topic === "browser-search-engine-modified" && data === "engine-default" && this.store.getState().Prefs.values[NO_DEFAULT_SEARCH_TILE_EXP_PREF]) {
+    if (topic === "browser-search-engine-modified" && data === "engine-current" && this.store.getState().Prefs.values[NO_DEFAULT_SEARCH_TILE_EXP_PREF]) {
+      this._currentSearchHostname = getShortURLForCurrentSearch();
       this.refresh({broadcast: true});
```

Comment:
Is this intended to update the current page? From testing, I have twitter as a top frecent site then from the search box, cmd-down to change default search to twitter, and it disappears immediately.

Reply:
yes, that's intended